catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/CswSubscriptionConfigFactory.java

```diff
@@ -0,0 +1,102 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint.event;
+
+import java.io.StringReader;
+
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.Unmarshaller;
+
+import org.codice.ddf.spatial.ogc.csw.catalog.endpoint.CswQueryFactory;
+import org.codice.ddf.spatial.ogc.csw.catalog.endpoint.CswSubscriptionEndpoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.opengis.cat.csw.v_2_0_2.GetRecordsType;
+
+public class CswSubscriptionConfigFactory {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CswSubscriptionConfigFactory.class);
+
+    /**
+     * The ManagedServiceFactory PID for the subscription's callback web service adapter,
+     * used to dynamically instantiate the web service's adapter
+     */
+    public static final String FACTORY_PID = "CSW_Subscription";
+
+    public static final String SUBSCRIPTION_ID = "subscriptionId";
+
+    public static final String FILTER_XML = "filterXml";
+
+    public static final String DELIVERY_METHOD_URL = "deliveryMethodUrl";
+
+    public static final String SUBSCRIPTION_UUID = "subscriptionUuid";
+
+    private String filterXml;
+
+    private String subscriptionId;
+
+    private String deliveryMethodUrl;
+
+    private CswSubscriptionEndpoint subscriptionService;
+
+    public CswSubscriptionConfigFactory(CswSubscriptionEndpoint subscriptionService) {
+        this.subscriptionService = subscriptionService;
+    }
+
+    private void restore() {
+        if (isComplete()) {
+            try {
+                Unmarshaller unmarshaller = CswQueryFactory.getJaxBContext()
+                        .createUnmarshaller();
+                StringReader sr = new StringReader(filterXml);
+                JAXBElement<GetRecordsType> jaxbElement =
+                        (JAXBElement<GetRecordsType>) unmarshaller.unmarshal(sr);
+                GetRecordsType request = jaxbElement.getValue();
+                if (!subscriptionService.hasSubscription(subscriptionId)) {
+                    subscriptionService.addOrUpdateSubscription(request, false);
+                }
+            } catch (Exception e) {
+                LOGGER.error("Error restoring subscription: {} with delivery URL: {} XML: {}",
+                        new Object[] {subscriptionId, deliveryMethodUrl, filterXml},
+                        e);
+            }
+        }
+    }
+
+    /**
```

Comment:
Can't we use the `init-method` attribute in the blueprint file instead of having to check that all values have been injected before proceeding?